WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/PostListAdapter.kt

```diff
@@ -186,75 +170,89 @@ class PostListAdapter(
-            }
+                showFeaturedImage(post, holder.featuredImage)
 
-            if (UploadService.isPostUploading(post)) {
-                holder.disabledOverlay.visibility = View.VISIBLE
-                holder.progressBar.isIndeterminate = true
-            } else if (!AppPrefs.isAztecEditorEnabled() && UploadService.isPostUploadingOrQueued(post)) {
-                // Editing posts with uploading media is only supported in Aztec
-                holder.disabledOverlay.visibility = View.VISIBLE
-            } else {
-                holder.progressBar.isIndeterminate = false
-                holder.disabledOverlay.visibility = View.GONE
-            }
+                // local drafts say "delete" instead of "trash"
+                if (post.isLocalDraft) {
+                    holder.date.visibility = View.GONE
+                    holder.trashButton.buttonType = PostListButton.BUTTON_DELETE
+                } else {
+                    holder.date.text = PostUtils.getFormattedDate(post)
+                    holder.date.visibility = View.VISIBLE
+                    holder.trashButton.buttonType = PostListButton.BUTTON_TRASH
+                }
 
-            updateStatusTextAndImage(holder.status, holder.statusImage, post)
-            updatePostUploadProgressBar(holder.progressBar, post)
-            configurePostButtons(holder, post)
-        }
+                if (UploadService.isPostUploading(post)) {
+                    holder.disabledOverlay.visibility = View.VISIBLE
+                    holder.progressBar.isIndeterminate = true
+                } else if (!AppPrefs.isAztecEditorEnabled() && UploadService.isPostUploadingOrQueued(post)) {
+                    // Editing posts with uploading media is only supported in Aztec
+                    holder.disabledOverlay.visibility = View.VISIBLE
+                } else {
+                    holder.progressBar.isIndeterminate = false
+                    holder.disabledOverlay.visibility = View.GONE
+                }
 
-        // load more posts when we near the end
-        if (position >= posts.size - 1 && position >= PostListFragment.POSTS_REQUEST_COUNT - 1) {
-            onLoadMoreListener?.onLoadMore()
-        }
+                updateStatusTextAndImage(holder.status, holder.statusImage, post)
+                updatePostUploadProgressBar(holder.progressBar, post)
+                configurePostButtons(holder, post)
+            }
 
-        holder.itemView.setOnClickListener {
-            onPostSelectedListener?.onPostSelected(post)
+            holder.itemView.setOnClickListener {
+                onPostSelectedListener?.onPostSelected(post)
+            }
         }
     }
 
-    private fun showFeaturedImage(postId: Int, imgFeatured: ImageView) {
-        val imageUrl = featuredImageUrls.get(postId)
+    private fun showFeaturedImage(post: PostModel, imgFeatured: ImageView) {
+        var imageUrl: String? = null
+        if (post.featuredImageId != 0L) {
```

Comment:
These actions should be taken in the bg thread, an issue is created for this #8474.